bundles/org.openhab.binding.bmwconnecteddrive/src/main/java/org/openhab/binding/bmwconnecteddrive/internal/handler/VehicleHandler.java

```diff
@@ -0,0 +1,826 @@
+                            }
+                        }
+                        updateChannel(CHANNEL_GROUP_VEHICLE_IMAGE, IMAGE_VIEWPORT, StringType.valueOf(newViewport));
+                    }
+                }
+                if (command instanceof DecimalType) {
+                    if (command instanceof DecimalType) {
+                        int newImageSize = ((DecimalType) command).intValue();
+                        if (channelUID.getIdWithoutGroup().equals(IMAGE_SIZE)) {
+                            synchronized (imageProperties) {
+                                if (imageProperties.size != newImageSize) {
+                                    imageProperties = new ImageProperties(imageProperties.viewport, newImageSize);
+                                    imageCache = Optional.empty();
+                                    proxy.ifPresent(prox -> prox.requestImage(config, imageProperties, imageCallback));
+                                }
+                            }
+                        }
+                        updateChannel(CHANNEL_GROUP_VEHICLE_IMAGE, IMAGE_SIZE, new DecimalType(newImageSize));
+                    }
+                }
+            });
+        } else if (CHANNEL_GROUP_DESTINATION.equals(group)) {
+            if (command instanceof StringType) {
+                int index = Converter.getIndex(command.toFullString());
+                if (index != -1) {
+                    selectDestination(index);
+                } else {
+                    logger.debug("Cannot select Destination index {}", command.toFullString());
+                }
+            }
+        } else if (CHANNEL_GROUP_SERVICE.equals(group)) {
+            if (command instanceof StringType) {
+                int index = Converter.getIndex(command.toFullString());
+                if (index != -1) {
+                    selectService(index);
+                } else {
+                    logger.debug("Cannot select Service index {}", command.toFullString());
+                }
+            }
+        } else if (CHANNEL_GROUP_CHECK_CONTROL.equals(group)) {
+            if (command instanceof StringType) {
+                int index = Converter.getIndex(command.toFullString());
+                if (index != -1) {
+                    selectCheckControl(index);
+                } else {
+                    logger.debug("Cannot select CheckControl index {}", command.toFullString());
+                }
+            }
+        } else if (CHANNEL_GROUP_CHARGE.equals(group)) {
+            handleChargeProfileCommand(channelUID, command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        callbackCounter = Optional.of(new ArrayList<ResponseCallback>());
+        updateStatus(ThingStatus.UNKNOWN);
+        final VehicleConfiguration config = getConfigAs(VehicleConfiguration.class);
+        if (config != null) {
+            configuration = Optional.of(config);
+            scheduler.execute(() -> {
```

Comment:
Not really blocking but quite some code with constructors and initialization is executed and also one critical section inside. Wanted to be sure to execute this asynchronous. But no problem to make it synchronous. Do you have an advice?

Reply:
Constructors and initialization can be executed synchronously. Critical sections cannot be guarded by executing them asynchronously. These should be synchronized.

Reply:
Ok, critical section is already synchronized - asnyc execution just in case if it's blocked so I don't want to delay other initializations. But seems acceptable so I'll remoce async init.